pkg/kubelet/apis/config/types.go

```diff
@@ -221,6 +221,11 @@ type KubeletConfiguration struct {
 	// CPUManagerPolicy is the name of the policy to use.
 	// Requires the CPUManager feature gate to be enabled.
 	CPUManagerPolicy string
+	// CPUManagerPolicyOptions allows to set extra options to fine tune the behaviour
+	// of the cpu manager policies.
+	// These options will be ignored by the "none" policy.
```

Comment:
shouldn't the provided options be validated by the selected policy, and the "none" policy fail that validation if options are provided?

Reply:
This is a good point. I made the `none` policy silently ignore everything because in the current incarnation it always succeed.
IMO giving options to the `none` policy is not wrong, it's just useless, hence the policy itself should ignore the options. But we were burned recently by a case on which we just logged and kept going, so I'm leaning towards making the policy fail if any options are given.

Reply:
> giving options to the none policy is not wrong, it's just useless

It seems like if we encounter that, the person configuring the kubelet made a mistake (typoed the cpuManager config field or something)... I think we should fail fast/loudly if we encounter configurations that make no sense in new fields we add

Reply:
Yes, I think this approach is better and I changed the code accordingly.

Reply:
it now looks like: https://github.com/kubernetes/kubernetes/pull/101432/commits/2108e23c90e6f0f020ed5c1d8a970fe025f3a84d#diff-c11864857fa45d5c2d29c3a85d753ccd2db31cca1c7fa046b7144e71f6e59029R152 we will see during the review of PR 101432 if we want to move this guard inside the `none` policy proper or if can be kept here.